src/main/java/com/example/resourcemanagerapp/validators/ResourceApiParamsValidator.java

```diff
@@ -0,0 +1,57 @@
+package com.example.resourcemanagerapp.validators;
+
+
+import com.example.resourcemanagerapp.additionalTypes.ResourceType;
+import com.example.resourcemanagerapp.exceptions.ApplicationException;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class ResourceApiParamsValidator {
+    public static void validateAddResourceParameters(String name, Integer userId, ResourceType type, String metadata) {
+        if(name == null || name.isEmpty()){
+            throw new ApplicationException("Name was not given!");
+        }
+        if(!StringValidator.checkIfContainsOnlyLettersAndDigits(name)){
+            throw new ApplicationException("Resource name should consists only of letters and digits!");
+        }
+        UserApiParamsValidator.validateUserId(userId);
+        validateResourceType(type);
+        if(metadata == null || metadata.isEmpty()){
+            throw new ApplicationException("Metadata was not given!");
+        }
+        validateMetadata(metadata);
+    }
+
+    public static void validateUpdateResourceNameParameters(Integer id, String newName){
+        validateResourceId(id);
+        if(!StringValidator.checkIfContainsOnlyLettersAndDigits(newName)) {
+            throw new ApplicationException("Resource name should consists only of letters and digits!");
+        }
+    }
+
+    public static void validateResourceId(Integer id){
+        if(id <= 0) {
+            throw new ApplicationException("Given resource id is invalid!");
+        }
+    }
+
+    public static void validateUpdateResourceMetadataParameters(Integer id, ResourceType type, String metadata){
+        validateResourceId(id);
+        validateResourceType(type);
+        validateMetadata(metadata);
+    }
+
+    public static void validateResourceType(ResourceType type){
```

Comment:
Można by tu też sprawdzać, oprócz nieuzupełnionego typu, czy dany typ pokrywa się z tymi w aplikacji.

Reply:
@karolk92 kazał mi usunąć sprawdzenie, czy typ pokrywa się z tym w aplikacji. Stwierdził, że lepsze jest przyjęcie enuma jako parametr zamiast Stringa, bo Spring automatycznie sprawdzi, czy taki typ istnieje, a jeśli nie, to wyrzuci wyjątek